File: python/folderprogram/ProdukElektronik.py
# ProdukElektronik.py

class ProdukElektronik:
    def __init__(self, kode, nama, harga, deskripsi):
        # FIXED: Menambahkan atribut 'kode'
        self.kode = kode
        self.nama = nama
        self.harga = harga
        self.deskripsi = deskripsi

    def display_info(self):
        print(f"Kode: {self.kode}")
        print(f"Nama: {self.nama}")
        print(f"Harga: Rp{self.harga:,.2f}")
        print(f"Deskripsi: {self.deskripsi}")

    # FIXED: Getter dan Setter dipindahkan ke dalam kelas
    def get_kode(self):
        return self.kode

    def set_kode(self, kode):
        self.kode = kode
        
    def get_nama(self):
        return self.nama

    def set_nama(self, nama):
        self.nama = nama

    def get_harga(self):
        return self.harga

    def set_harga(self, harga):
        self.harga = harga

    def get_deskripsi(self):
        return self.deskripsi

    def set_deskripsi(self, deskripsi):
        self.deskripsi = deskripsi

File: php/folderprogram/ProdukElektronik.php
<?php
class ProdukElektronik {
    private $kode;
    private $nama;
    private $harga;
    private $deskripsi;

    public function __construct($kode = "", $nama = "", $harga = 0, $deskripsi = "") {
        $this->kode = $kode;
        $this->nama = $nama;
        $this->harga = $harga;
        $this->deskripsi = $deskripsi;
    }

    // Getter & Setter
    public function getKode() { return $this->kode; }
    public function setKode($kode) { $this->kode = $kode; }

    public function getNama() { return $this->nama; }
    public function setNama($nama) { $this->nama = $nama; }

    public function getHarga() { return $this->harga; }
    public function setHarga($harga) { $this->harga = $harga; }

    public function getDeskripsi() { return $this->deskripsi; }
    public function setDeskripsi($deskripsi) { $this->deskripsi = $deskripsi; }

    public function displayInfo() {
        echo "Kode: {$this->kode}<br>";
        echo "Nama: {$this->nama}<br>";
        echo "Harga: Rp{$this->harga}<br>";
        echo "Deskripsi: {$this->deskripsi}<br>";
    }
}
?>
